kong/utils.go

```diff
@@ -797,32 +799,168 @@ func fillConfigRecordDefaultsAutoFields(plugin *Plugin, schema map[string]interf
+		}
+	}
+
 	if plugin.Protocols == nil {
 		plugin.Protocols = getDefaultProtocols(gjsonSchema)
 	}
 	if plugin.Enabled == nil {
 		plugin.Enabled = Bool(true)
 	}
+
+	return nil
+}
+
+func getDefaultPartialPath(partialLinks []*PartialLink, pluginSchema gjson.Result, partials []*Partial) error {
+	supportedPartials := pluginSchema.Get("supported_partials")
+	if !supportedPartials.Exists() {
+		return fmt.Errorf("schema does not contain supported_partials")
+	}
+
+	// internal function to find a partial by its ID
+	// this is useful to get partial's type for finding
+	// its default path in the config
+	findPartialByID := func(partialID string) *Partial {
+		for _, p := range partials {
+			if *p.ID == partialID {
+				return p
+			}
+		}
+		return nil
+	}
+
+	for _, p := range partialLinks {
+		if p.Path != nil {
+			continue
+		}
+
+		partialNeeded := findPartialByID(*p.ID)
+		if partialNeeded == nil {
+			return fmt.Errorf("partial with ID %s not found", *p.ID)
+		}
+
+		defaultPartialPathInSchema := supportedPartials.Get(*partialNeeded.Type)
+		if !defaultPartialPathInSchema.Exists() {
+			return fmt.Errorf("schema does not contain default partial path for partial type %s", *partialNeeded.Type)
+		}
+		defaultPartialPathArray := defaultPartialPathInSchema.Array()
+		if len(defaultPartialPathArray) > 1 {
+			return fmt.Errorf(">1 supported paths found for partial type %s; provide a path in config", *partialNeeded.Type)
+		}
+
+		p.Path = String(defaultPartialPathArray[0].String())
+	}
+
+	return nil
+}
+
+func fillPartialConfigsInPlugin(plugin *Plugin, partials []*Partial) (Configuration, error) {
+	config := plugin.Config.DeepCopy()
+
+	// internal function to find a partial by its ID
+	// this is useful to get partial's type for finding
+	// its default path in the config
+	findPartialByID := func(partialID string) *Partial {
+		for _, p := range partials {
+			if *p.ID == partialID {
+				return p
+			}
+		}
+		return nil
+	}
+
+	const configString string = "config."
+
+	for _, partialLink := range plugin.Partials {
+		sanitisedPath := strings.ReplaceAll(*partialLink.Path, configString, "")
+		partialNeeded := findPartialByID(*partialLink.ID)
+		if partialNeeded == nil {
+			return nil, fmt.Errorf("partial with ID %s not found", *partialLink.ID)
+		}
+		// Gateway does not support overrides yet.
+		// Thus, we are adding the config directly
+		config[sanitisedPath] = partialNeeded.Config
+	}
+
+	return config, nil
+}
+
+func fillConfigRecordWithPartialsConfig(plugin *Plugin, pluginSchema map[string]interface{},
+	partials []*Partial,
+) error {
+	jsonb, err := json.Marshal(&pluginSchema)
+	if err != nil {
+		return err
+	}
+	gjsonSchema := gjson.ParseBytes(jsonb)
+
+	if plugin.Config == nil {
+		plugin.Config = make(Configuration)
+	}
+
+	if len(partials) > 0 {
+		err = getDefaultPartialPath(plugin.Partials, gjsonSchema, partials)
+		if err != nil {
+			return err
+		}
+		plugin.Config, err = fillPartialConfigsInPlugin(plugin, partials)
```

Comment:
Nit: Since this func is only called where `plugin.Config` is assigned to the return value, why don't we just perform the assignment in `fillPartialConfigsInPlugin`?

Specifically, what I'm suggesting is to drop the first return value, make the assignment inside the `fillPartialConfigsInPlugin` and potentially do away with the deep copy (I _think_ it won't be necessary at that point).

WDYT?